### ISSUE TITLE
Error when using Resend in Remix Run

### ISSUE THREAD

**Issue by hasinitharupaba** [resend/resend-node#219]
### Description
I have built a Remix web app and I'm trying to implement Resend for email functionality. However, I'm encountering an error in my console:

resend.js:18 Uncaught ReferenceError: process is not defined at node_modules/resend/build/src/resend.js
![Screenshot 2023-09-19 at 2 39 29 AM](https://github.com/resendlabs/resend-node/assets/83213422/5681333a-4971-4052-9756-44fd7ffa09ea)

I am using resend version ^1.1.0 and remix version 2.

This is the code where I am trying to use it.

```// contact.tsx (contact page)

import { Alert, AlertDescription, AlertTitle } from "~/components/ui/alert";
import DataForm from "~/components/atoms/DataForm";
import type { ActionArgs } from "@remix-run/node";
import { redirect } from "@remix-run/node";
import { Resend } from "resend";
import { useSearchParams } from "react-router-dom";
import EmailTemplate from "~/components/atoms/email-template";
import { Link } from "@remix-run/react";
import Button from "~/components/atoms/Button";

export const action = async ({ request }: ActionArgs) => {
  if (process) {
    const resend = new Resend(process.env.RESEND_API_KEY);
    if (request.method === "POST") {
      const body = await request.formData();
      const data = Object.fromEntries(body);

      const res: any = await resend.emails.send({
        from: "Spiced.Tech <no-reply@xxx.com>",
        to: ["h@xxx.com"],
        subject: `Contact form submission - ${new Date().toDateString()}`,
        react: EmailTemplate({
          name: data.name,
          email: data.email,
          existingWebsite: data.existingWebsite,
          webUrl: data.webUrl,
          numberOfPages: data.numberOfPages,
          webService: data.webService,
          nature: data.nature,
          message: data.message,
        }),
      });

      if (res.id) {
        return redirect(`/contact?status=success`);
      } else {
        return redirect(
          `/contact?status=failed&statusCode=${res.statusCode}&message=${res.message}`
        );
      }
    } else {
      return null;
    }
  }
};

export async function loader() {
  return null;
}

export default function Contact() {
  const [searchParams] = useSearchParams();
  const formStatus = searchParams.get("status");
  const formMessage = searchParams.get("message");

  return (
    <>
      <div className="right-[20px] hidden items-end justify-end md:relative md:flex">
        {formStatus === "success" && (
          <Alert variant="success">
            <AlertTitle>Success!</AlertTitle>
            <AlertDescription>We received your message.</AlertDescription>
          </Alert>
        )}
        {formStatus === "failed" && (
          <Alert variant="destructive">
            <AlertTitle>Oops!</AlertTitle>
            <AlertDescription>{formMessage}</AlertDescription>
          </Alert>
        )}
      </div>

      <DataForm />
    </>
  );
}

**Comment by jayanratna** [comment]
Hey @hasinitharupaba, looks like you're trying to access `process` in your client side code. The `process` variable is only accessible in your server side code.

**Comment by hasinitharupaba** [comment]
This is hitting directly from the resend node package and also my page is already server-side.

**Comment by kestrelgorlick** [comment]
I am having this same issue, and my error is originating from within `node_modules` as well, not from my code but from the ResendJS package.

**Comment by omelhus** [comment]
You need to create a service file called "resend.server.ts" and export resend from there. That way Remix will know not to include Resend in the client bundle.

```typescript
// ~/services/resend.server.ts
import { Resend } from "resend";

export const resend = new Resend(process.env.RESEND_API_KEY!);
```

```typescript
import { Alert, AlertDescription, AlertTitle } from "~/components/ui/alert";
import DataForm from "~/components/atoms/DataForm";
import type { ActionArgs } from "@remix-run/node";
import { redirect } from "@remix-run/node";
import { resend } from "~/services/resend.server";
import { useSearchParams } from "react-router-dom";
import EmailTemplate from "~/components/atoms/email-template";
import { Link } from "@remix-run/react";
import Button from "~/components/atoms/Button";

export const action = async ({ request }: ActionArgs) => {
    if (request.method === "POST") {
      const body = await request.formData();
      const data = Object.fromEntries(body);
      const res: any = await resend.emails.send({
        from: "Spiced.Tech <no-reply@xxx.com>",
        to: ["h@xxx.com"],
[...]
```

No need to check for process in the action, and resend is instantiated in resend.server.ts.

**Comment by zenorocha** [comment]
This has been fixed on [v2.1.0](https://github.com/resend/resend-node/releases/tag/v2.1.0).
